mappings/net/minecraft/client/gui/GuiMainMenu.mapping

```diff
@@ -2,11 +2,23 @@ CLASS bgu net/minecraft/client/gui/GuiMainMenu
 	FIELD A SPLASHES_LOC Lkp;
 	FIELD B LOGO_LOC Lkp;
 	FIELD C PANORAMA_LOC [Lkp;
+	FIELD D outdatedGlTextWidth I
```

Comment:
Uh they're private to GuiMainMenu, and in the ctor it's explicitly set to = the static field `OUTDATED_GL_TEXT`. 
I *guess* its not hardcoded /per se/ but in the current version (16w38a) it's only ever going to be GL warnings. I think at some point this same code was used for the warning when they were dropping support for PPC, so I understand your point. I guess generic warning is probably fine, it was just the hardcoded `= I18n.translate...` that got me.

```java
    public GuiMainMenu() {
        ...
        this.outdatedGlText = GuiMainMenu.OUTDATED_GL_TEXT;
        ...
        if (!GLContext.getCapabilities().OpenGL20 && !GlExtensionHandler.canUseShaders()) {
            this.outdatedGlTitle = I18n.translate("title.oldgl1", new Object[0]);
            this.outdatedGlText = I18n.translate("title.oldgl2", new Object[0]);
            this.outdatedGlLink = "https://help.mojang.com/customer/portal/articles/325948?ref=game";
        }
        ...
    }
    ...
    public void init() {
        ...
        synchronized (this.mutex) {
            this.outdatedGlTitleWidth = this.fontRenderer.getStringWidth(this.outdatedGlTitle);
            this.outdatedGlTextWidth = this.fontRenderer.getStringWidth(this.outdatedGlText);
            final int v4 = Math.max(this.outdatedGlTitleWidth, this.outdatedGlTextWidth);
            this.outdatedGlAlignLeft = (this.width - v4) / 2;
            this.outdatedGlAlignTop = this.widgets.get(0).y - 24;
            this.outdatedGlAlignRight = this.outdatedGlAlignLeft + v4;
            this.outdatedGlAlignBottom = this.outdatedGlAlignTop + 24;
        }
        ...
    }
    ...
```